ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckOperation.java

```diff
@@ -75,7 +78,25 @@ public int execute() throws HiveException, IOException, TException, MetastoreExc
       MsckInfo msckInfo = new MsckInfo(SessionState.get().getCurrentCatalog(), tableName.getDb(), tableName.getTable(),
           desc.getFilterExp(), desc.getResFile(), desc.isRepairPartitions(),
           desc.isAddPartitions(), desc.isDropPartitions(), partitionExpirySeconds);
-      return msck.repair(msckInfo);
+      int result = msck.repair(msckInfo);
+      if (msckInfo.getSmallFilesStats() != null && !msckInfo.getSmallFilesStats().isEmpty()){
+        List<String> logInfo = new ArrayList<>(msckInfo.getSmallFilesStats().size());
+        for (Map.Entry<String, String> me : msckInfo.getSmallFilesStats().entrySet()){
+          String partitionName = me.getKey();
+          String smallFilesStats = me.getValue();
+          logInfo.add("This table/partition average file size is less than Hive average file size.\n " +
+                  "The partition name is " + partitionName + ". " + smallFilesStats);
+        }
+        SessionState ss = SessionState.get();
+        if (ss != null && ss.getConsole() != null) {
+          ss.getConsole().printInfo("[MSCK] Small files detected.");
+          for (String line : logInfo) {
+            ss.getConsole().printInfo("[MSCK] " + line);
+          }
+        }
+        LOG.info("There are small files exist.\n{}", String.join("\n", logInfo));
```

Comment:
should this be under warn?

Reply:
The text warnings are printed to the end user in the console.
So this doesn't have to be warnings in the LOG, and small files are common to see.